### PR TITLE
fix(Tooltip): prevent overriding the `aria-label` attribute

### PR DESCRIPTION
**Pull request by snitin315**
Fixes https://github.com/razorpay/blade/issues/1386

The issue exists on the web only because we don't set the `accessibilityLabel` prop to trigger on react-native.

https://github.com/razorpay/blade/blob/7c116d54164124651fe420a93f182fb4caf49cf9/packages/blade/src/components/Tooltip/Tooltip.native.tsx#L72-L80

### CODE REVIEW

**Review by anuraghazra — APPROVED** [review]

File: packages/blade/src/components/Tooltip/__tests__/Tooltip.web.test.tsx [hunk 1]
```diff
@@ -53,7 +53,7 @@ describe('<Tooltip />', () => {
 
   it('should render tooltip with custom zIndex', () => {
     const buttonText = 'Hover me';
-    const { container, getByRole, queryByRole } = renderWithTheme(
+    const { baseElement, getByRole, queryByRole } = renderWithTheme(
```
Comment (saurabhdaware) [inline packages/blade/src/components/Tooltip/__tests__/Tooltip.web.test.tsx:56]: expected change?

File: packages/blade/src/components/Tooltip/Tooltip.web.tsx [hunk 2]
```diff
@@ -98,7 +98,7 @@ const Tooltip = ({
       {React.cloneElement(children, {
         ref: refs.setReference,
         ...mergeProps(children.props, getReferenceProps()),
-        ...makeAccessible({ label: content }),
+        ...makeAccessible({ describedBy: content }),
```
Comment (saurabhdaware) [inline packages/blade/src/components/Tooltip/Tooltip.web.tsx:101]: did you mean `description` here? `describedby` expects id of the description no?

Also why this change?

image

https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-describedby#example

File: packages/blade/src/components/Tooltip/__tests__/__snapshots__/Tooltip.web.test.tsx.snap [hunk 3]
```diff
@@ -175,7 +186,7 @@ exports[`<Tooltip /> should render 1`] = `
 <body>
   <div>
     <button
-      aria-describedby=":r0:"
+      aria-describedby="Hello world"
```
Comment (saurabhdaware) [inline packages/blade/src/components/Tooltip/__tests__/__snapshots__/Tooltip.web.test.tsx.snap:189]: `:r0:` would be expected no?

File: packages/blade/src/components/Tooltip/__tests__/Tooltip.web.test.tsx (hunk 1 above)
Comment (snitin315) [reply packages/blade/src/components/Tooltip/__tests__/Tooltip.web.test.tsx:56]: Yes, we want to snapshot the open tooltip and not the trigger.

File: packages/blade/src/components/Tooltip/Tooltip.web.tsx (hunk 2 above)
Comment (anuraghazra) [reply packages/blade/src/components/Tooltip/Tooltip.web.tsx:101]: Oh right, it should not be the content

Comment (snitin315) [reply packages/blade/src/components/Tooltip/Tooltip.web.tsx]: Yeah, I updated the logic not to override the aria-label if provided by the user. We're not adding `aria-description` as [recommended in the docs.](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Attributes/aria-description)

@anuraghazra @saurabhdaware

**Review by anuraghazra — APPROVED** [review]

**Review by saurabhdaware — APPROVED** [review]
